src/app/modules/onboarding/pages/nonhd-wallet-wizard/nonhd-wallet-wizard.component.ts

```diff
@@ -99,29 +102,67 @@ export class NonhdWalletWizardComponent implements OnInit, OnDestroy {
         this.keystoresFormGroup.markAllAsTouched();
         break;
     }
-    if (this.keystoresFormGroup.valid){
+    if (this.keystoresFormGroup.valid && !this.slashingProtection?.invalid){
       this.stepper?.next();
     }
   }
 
+  private get importKeystores$(): Observable<any>{
+    const keystoresImported: string[] = [];
+    let keystorePasswords: string[] = [];
+    (this.keystoresFormGroup.controls['keystoresImported'] as FormArray).controls.forEach((keystore: AbstractControl) => {
+      keystoresImported.push(JSON.stringify(keystore.get('keystore')?.value));
+      keystorePasswords.push(keystore.get('keystorePassword')?.value);
+    });
+    if(this.importAccounts?.uniqueToggleFormControl.value){
```

Comment:
Might be worthwhile adding a comment about the complex observable logic this function is doing. We can just explain why we're doing a list of requests

Reply:
added some comments